general-concepts/ebuild-revisions/text.xml

```diff
@@ -8,37 +8,99 @@
 Ebuilds may have a Gentoo revision number associated with them. This is a
 <c>-rX</c> suffix, where <c>X</c> is an integer <d/> see <uri
 link="::ebuild-writing/file-format#File Naming Rules"/>. This
-component must only be used for Gentoo changes, not upstream releases. By
-default, <c>-r0</c> is implied.
+component must only be used for Gentoo changes, not upstream releases.
+An ebuild with no explicit revision number has the implicit <c>-r0</c>
+revision.
 </p>
 
-<p>
-Ebuilds should have their <c>-rX</c> incremented whenever a change is made which
-will make a substantial difference to what gets installed by the package <d/> by
-substantial, we generally mean "something for which many users would want to
-upgrade". This is usually for bugfixes.
-For any such revision bump, the new ebuild should be based on the
-previous revision to ensure that fixes aren't dropped accidentally.
-</p>
+<p>Ebuild revisions usually serve two purposes:</p>
+<ol>
+  <li>
+    keeping an older copy of an ebuild around when doing a potentially
+    breaking change, and
+  </li>
+
+  <li>
+    propagating the rebuild of a package when performing a meaningful
+    change that would otherwise go unnoticed by users who have installed
+    the current version already.
+  </li>
+</ol>
 
 <p>
-Simple compile fixes do <b>not</b> warrant a revision bump; this is because they do
-not affect the installed package for users who already managed to compile it.
-Small documentation fixes are also usually not grounds for a new revision.
-In particular, this applies if the package has a substantial compilation
-time; developers should use their best judgement in these circumstances.
+Developers are encouraged to use common sense when determining
+whether to introduce a new <c>-rX</c> revision. The following rule
+of thumb could be used as a guideline:
 </p>
+<ol>
+  <li>
+    If the change can cause the package to be broken to the point
+    of requiring users to revert to the previous version (in the case
+    of packages marked stable, every non-trivial change is classified
+    as such), then a new revision should be introduced and the old one
+    kept. If the package has stable keywords, the new revision should
+    be dropped to <c>~arch</c> (see
+    <uri link="::keywording#Keywording on Upgrades"/>).
+  </li>
 
-<important>
-For ebuilds marked stable on at least one arch, only trivial edits can be made
-without a bump (e.g. typo fixes in elog messages). Even simple changes may
-result in a breakage. <b>Modifying stable ebuilds should be avoided.</b>
-</important>
+  <li>
+    If the change makes a substantial difference to the user who already
+    installed the package (fixes runtime issues, changes installed files,
+    etc.) and it would not be propagated using other means, then
+    the ebuild should be renamed to a new revision. If the package has
+    stable keywords, they should be kept intact.
```

Comment:
When you say `they should be kept intact`, are you suggesting committing straight to stable as part of the rename? Or did you want to say `it should be kept intact`, meaning that the stable ebuild should be left alone and a revbump should be made with all ~arch. Also, `package` has stable keywords, or `ebuild` has stable keywords?

Reply:
The keywords should be kept intact, i.e. straight to stable.

Ebuild is a kind of package. In this context, they're synonymous since other kind of packages don't apply here.

Reply:
This is not gonna fly I'm afraid:

```
$ git mv ffmpeg-3.3.3{,-r1}.ebuild
$ repoman
[...]
  KEYWORDS.stable [fatal]       1
   media-video/ffmpeg/ffmpeg-3.3.3-r1.ebuild added with stable keywords: amd64 arm ia64 x86
```

Reply:
There's `-S`/`--straight-to-stable` option for exactly that purpose, for over two years now.

Reply:
it's probably worth mentioning it then; I didn't know about it and would have used --force